### PR TITLE
feat: category and user profile page

### PR DESCRIPTION
**Pull request by bhtibrewal**
Feat: created checkout and user profile page 
Done: login/logout/signup, update user date

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -38,10 +40,11 @@ function App() {
               </ProductFilterProvider>
             }
           />
-        
+          <Route path="/user_profile" element={<UserProfile />} />
```
Comment (yash9263) [inline src/App.jsx:43]: use `user-profile` in routes

**Review by yash9263 — APPROVED** [review]

**Comment by yash9263** [comment]
made some comments you can improve code looks nice and clean

File: src/custom_hooks/useUserData.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      default:
+        return { ...state };
+    }
+  };
+
+  const [userData, userDataDispatch] = useReducer(
+    user_data_reducer,
+    initialUserData
+  );
+  console.log(userData);
```
Comment (sri-rishi) [inline src/custom_hooks/useUserData.jsx:40]: Suggestion: remove `console.log` before committing

File: src/pages/auth/sign_up/SignUp.jsx [hunk 3]
```diff
@@ -30,8 +32,8 @@ export const SignUp = () => {
     return agreeToTerms && canSubmit();
   };
   return (
-    <main className="main flex-col">
-      <form className="flex-col signup-sec">
+    <main className="main flex-align-center">
+      <div className="flex-col signup-sec">
```
Comment (sri-rishi) [inline src/pages/auth/sign_up/SignUp.jsx:36]: Suggestion: can use `form`  tag  here for better semantic

**Review by sri-rishi — APPROVED** [review]
Code looks good, just mention some change , feel free to merge

File: src/App.jsx (hunk 1 above)
Comment (bhtibrewal) [reply src/App.jsx:43]: Okk will do that

**Pull request merged**